include/sst/voice-effects/filter/FiltersPlusPlus.h

```diff
@@ -0,0 +1,478 @@
+/*
+ * sst-effects - an open source library of audio effects
+ * built by Surge Synth Team.
+ *
+ * Copyright 2018-2023, various authors, as described in the GitHub
+ * transaction log.
+ *
+ * sst-effects is released under the GNU General Public Licence v3
+ * or later (GPL-3.0-or-later). The license is found in the "LICENSE"
+ * file in the root of this repository, or at
+ * https://www.gnu.org/licenses/gpl-3.0.en.html
+ *
+ * The majority of these effects at initiation were factored from
+ * Surge XT, and so git history prior to April 2023 is found in the
+ * surge repo, https://github.com/surge-synthesizer/surge
+ *
+ * All source in sst-effects available at
+ * https://github.com/surge-synthesizer/sst-effects
+ */
+
+#ifndef INCLUDE_SST_VOICE_EFFECTS_FILTERSPLUSPLUS_H
+#define INCLUDE_SST_VOICE_EFFECTS_FILTERSPLUSPLUS_H
+
+#include "sst/basic-blocks/params/ParamMetadata.h"
+
+#include "sst/filters++.h"
+
+#include "../VoiceEffectCore.h"
+
+#include <iostream>
+
+namespace sst::voice_effects::filter
+{
+template <typename VFXConfig, filtersplusplus::FilterModel Model>
+struct FiltersPlusPlus : core::VoiceEffectTemplateBase<VFXConfig>
+{
+    static constexpr const char *effectName{"Filters++"};
+
+    static constexpr int numFloatParams{4};
+    static constexpr int numIntParams{5};
+
+    enum FloatParams
+    {
+        fpCutoffL,
+        fpCutoffR,
+        fpResonance,
+        fpExtra
+    };
+
+    enum IntParams
+    {
+        ipStereo,
+        ipPassband,
+        ipSlope,
+        ipDrive,
+        ipSubmodel
+    };
+
+    FiltersPlusPlus() : core::VoiceEffectTemplateBase<VFXConfig>()
+    {
+        std::fill(priorFP.begin(), priorFP.end(), -1000.f);
+        std::fill(priorIP.begin(), priorIP.end(), -1);
+
+        filter.init();
+        filter.setFilterModel(Model);
+
+        using fmd = filtersplusplus::FilterModel;
+
+        if constexpr (Model == fmd::VemberClassic)
```

Comment:
Oh this really looks wrong. Why are you doing this with actual enum values here rather than dynamically building it with the API?